controllers/storage.go

```diff
@@ -353,6 +354,33 @@ func (s *Storage) CopyFromPath(artifact *sourcev1.Artifact, path string) (err er
 	return s.Copy(artifact, f)
 }
 
+// CopyToPath copies the contents of the given atrifact to the path.
+func (s *Storage) CopyToPath(atrifact *sourcev1.Artifact, path string) error {
```

Comment:
Have you thought about the behavior when the target path destination already exists? Would a force overwrite toggle make sense for this situation?

Reply:
My personal opinion is that there should always be an overwrite from the target path and everything below it. Doing weird merging will be very difficult for users to debug and require them to jump into source-controller to check the contents of the generated artifact.

Reply:
I agree, though what I had in mind was about either erroring out or simply overwriting, didn't consider merging as an option really. For example, if the toggle would be disabled, users would see something like "Error: Destination path already exists".
No idea if users might find this useful or not to be honest.

Reply:
Lets keep it simple for now and come back to this is people ask for some other behavior.